desktop/window-handlers/external-links/index.js

```diff
@@ -103,7 +124,46 @@ module.exports = function( webContents ) {
 
 		const openUrl = format( parsedUrl );
 
-		log.info( `Using system default handler for URL: ${ openUrl }` );
 		openInBrowser( event, openUrl );
 	} );
+
+	ipc.on( 'cannot-open-editor', async( event, info ) => {
+		const { origin, editorUrl } = info;
+		const parsedURL = new URL( origin );
+
+		const promptWpAdminAuth = () => {
+			try {
+				const selected = dialog.showMessageBox( mainWindow, {
+					type: 'info',
+					buttons: [ 'Proceed in Browser', 'Cancel' ],
+					title: 'Jetpack Authorization Required',
+					message: 'This feature requires that Secure Sign-On is enabled in the Jetpack settings of the site:' +
+					'\n\n' +
+					`${parsedURL.origin}`,
+					detail: 'You may try again after changing the site\'s Jetpack settings ' +
+					'(requires application re-start), ' +
+					'or you can proceed in an external browser.'
+				} );
+
+				switch ( selected ) {
+					case 0:
+						log.info( 'User selected \'Proceed in Browser\'...' )
+						openInBrowser( event, editorUrl );
+						break;
+					case 1:
+						log.info( 'User selected \'Cancel\'...' );
+						break;
+				}
+				showMySites( mainWindow );
+				// Update last location so a redirect isn't automatically triggered on app relaunch
+				Settings.saveSetting( settingConstants.LAST_LOCATION, `/stats/day/${parsedURL.hostname}` );
+			} catch ( error ) {
+				log.error( 'Failed to prompt for Jetpack authorization: ', error );
+			}
+		}
+
+		await delay( 300 );
```

Comment:
can we leave an explanatory comment on why we're waiting and why we're waiting 300ms?

 - Is there an underlying race condition we're trying to avoid?
 - are we trying to let network requests settle?
 - are we only trying to avoid instantly popping up the window?

Reply:
Will leave a note. 

It works just fine without the delay, except if one of the buttons presented _without_ the [Gutenberg placeholder UI](https://github.com/Automattic/wp-calypso/blob/master/client/gutenberg/editor/controller.js#L110-L112) having a chance to display, dismissing the modal will leave the UI in a state where the Calypso "chrome" (masterbar, etc) disappears from the current view. It's really odd - I should take a video to explain what I mean. 

This just gives the placeholder UI a chance to display, so that things look as expected when dismissed.